### PR TITLE
Fix #306917 crash on tie in score with parts

### PR DESCRIPTION
**Pull request by tobik**
Resolves: https://musescore.org/en/node/306917

Added null pointer check

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://github.com/musescore/Documentation/blob/master/WorkflowAndGuidelines/CodeGuidelines.md)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [x] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [ ] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

File: libmscore/edit.cpp [hunk 1]
```diff
@@ -4829,13 +4829,13 @@ void Score::undoAddElement(Element* element)
                               sm = cr2->staffIdx() - cr1->staffIdx();
                         Chord* c1 = findLinkedChord(cr1, score->staff(staffIdx));
                         Chord* c2 = findLinkedChord(cr2, score->staff(staffIdx + sm));
-                        Note* nn1 = c1->findNote(n1->pitch(), n1->unisonIndex());
+                        Note* nn1 = c1 ? c1->findNote(n1->pitch(), n1->unisonIndex()) : 0;
```
Comment (Howard-C) [inline libmscore/edit.cpp:4832]: Please use `nullptr`, same for the rest. See the [code style documentation](https://github.com/musescore/Documentation/blob/master/WorkflowAndGuidelines/CodeGuidelines.md).

Comment (Jojo-Schmitz) [reply libmscore/edit.cpp:4832]: that's the new master code style though...

Comment (Jojo-Schmitz) [reply libmscore/edit.cpp:4832]: 0 is a perfectly acceptable value for a pointer

File: libmscore/edit.cpp [hunk 2]
```diff
@@ -4829,13 +4829,13 @@ void Score::undoAddElement(Element* element)
...
                         Note* nn2 = c2 ? c2->findNote(n2->pitch(), n2->unisonIndex()) : 0;
 
                         // create tie
                         Tie* ntie = toTie(ne);
                         ntie->eraseSpannerSegments();
-                        ntie->setTrack(c1->track());
+                        ntie->setTrack(c1 ? c1->track() : 0);
```
Comment (njvdberg) [inline libmscore/edit.cpp:4838]: Why should this track 0? I think the first question would be why is <code>c1</code> equal to <code>nullptr</code>.

**Comment by Jojo-Schmitz** [comment]
Please don't reference musesecore issue tracker issues with #nnnnnnn (as those may autoling to a PR or GitHub issue of that number), but via a direct link, in this case https://musescore.org/en/node/306917

File: libmscore/edit.cpp (hunk 1 above)
Comment (njvdberg) [inline libmscore/edit.cpp:4832]: Also here, the question is why <code>c1</code> is equal to <code>nullptr</code>. This is about linked parts, to the there should be a linked chord. Is <code>findLinkedChord</code> returning the correct <code>Chord</code>?

Comment (tobik) [reply libmscore/edit.cpp:4832]: I didn't want to make unnecessary changes (the line below) but still want to fit the new code to the existing one.

**Comment by tobik** [comment]
Changed

File: libmscore/edit.cpp (hunk 1 above)
Comment (tobik) [reply libmscore/edit.cpp:4832]: findLinkedChord(..) returns nullptr

Comment (Jojo-Schmitz) [reply libmscore/edit.cpp:4832]: Maybe that's the real bug?

Comment (njvdberg) [reply libmscore/edit.cpp:4832]: And I think that's the problem, for me it shouldn't. Apparently it return nullptr for note if the second staff but not on the first which doesn't make sense.

Comment (Jojo-Schmitz) [reply libmscore/edit.cpp:4832]: Then again using track 0 is a pretty safe bet, staff 0, voice 1 always exists. Maybe it should be `staffNo+1*VOICES` though?

Comment (njvdberg) [reply libmscore/edit.cpp:4832]: @Jojo-Schmitz I think it is. When working on linked parts I saw quite some place where linked parts and linked staves were mixed up. I'm wondering whether something similar happened too. Also, when looking for a linked chord, why not looking into the linked element list?

Comment (njvdberg) [reply libmscore/edit.cpp:4832]: But it is still a bet :-)
When it comes to linked part, the <code>QMultiMap<int, int> tracks</code> should be chcked to find the mapping between the tracks of master and part score.

**Pull request closed without merging**